Properly destroying the objects that contain daemonized threads

chmduquesne opened the pull request:
Consumer, Producer and ReentrantTimer all create daemonized threads. Problem: when destroying these objects, the daemonized threads are not destroyed and the threads live forever with no mean to reach them.

This created issue #323, which I could deal with by using stop() manually, but there is no reason to force the user to force the user to do that when it could be dealt with automatically.

This pull request fixes this.

dpkp commented:
can you submit pep8 fixes / ae0017cb0172b633a56084be6e4779fc0bfedac2 separately?

chmduquesne commented:
Yes.

Actually I did this commit to trigger a travis build on my clone and check
that the build was passing before doing a pull request (I don't know how
else to do).

I will remove the pep8ification.

chmduquesne commented:
I have removed the pep8 fix, and rebased my patch onto master, so that this can be merged with a fast forward.

Travis reports an error on this pull request, but I don't understand why: on my clone, [it does not complain](https://travis-ci.org/chmduquesne/kafka-python/builds/52819116). Can you relaunch the [incriminated build](https://travis-ci.org/mumrah/kafka-python/jobs/52819175)?

dpkp commented:
not sure what is going on w/ tests -- `test_batched_simple_producer__triggers_by_message` may need some investigation
